JoinColony/colonyCDapp
# Clean up and refactor `UserAvatar` component

## Description

This PR takes care of the cleaning and refactoring of the `UserAvatar` component.

**New stuff** ✨

* Added new lambda function `getReputationForTopDomains `
* Added `TokenInfoPopoverWrapper `
* Created a new  `UserReputationItem` component that handles each item in the user reputation list

**Changes** 🏗

* Cleaned up `UserAvatar` component
* Combined `UserInfoPopover` and `MemberInfoPopover` into one component
* Simplified the `Tag` component

**Deletions** ⚰️

* `InfoPopOver`
* `MemberInfoPopover`
* `UserInfoPopover`

## Testing

1. Add reputation using truffle to a user and a domain to show reputation in the UserInfoPopover:
```c = await IColony.at('0x3a157280ca91bB49dAe3D1619C55Da7F9D4438c3')```
```c.emitDomainReputationReward(1, accounts[0], '1000000000000000000')```

2. Make sure the reputation monitor is ON:
`http://127.0.0.1:3001/reputation/monitor/toggle`
`http://127.0.0.1:3001/reputation/monitor/status`

Without Reputation
<img width="564" alt="Screen Shot 2022-11-23 at 4 58 31 PM" src="https://user-images.githubusercontent.com/71563622/203653116-6ccb9444-3060-4028-a0b9-f98e122e8a59.png">

With Reputation
<img width="548" alt="Screen Shot 2022-11-23 at 4 55 10 PM" src="https://user-images.githubusercontent.com/71563622/203653100-ac7f4de6-e626-4bde-8c04-a87dada96144.png">


## TODO

- [x] Cleanup imports and remove what's no longer required
- [x] Remove all traces of IPFS / withDataFetcher logic
- [x] Remove the Hooked higher order function
- [x] Make sure styles are up to date with the latest ones from the Dapp
- [x] Introduce logic to account for thumbnails, as we now have that value in the db
- [x] Clean up, and wire the user tooltip
- [x] Fix the user avatar hover styles
- [x] Create task for the refactoring of getMainClasses in a separate [PR](https://github.com/JoinColony/colonyCDapp/issues/89)


Resolves #63
